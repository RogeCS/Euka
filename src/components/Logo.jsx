import React from "react";
import DarkLogo from "../assets/static/euka-dark-logo.svg";
import LightLogo from "../assets/static/euka-dark-logo.svg";

import "../styles/components/Logo.scss";

const Logo = ({ color }) => {
  return (
    <div className="logo">
      <a href="/" className="logo__link">
        <img
          className="logo__image"
          src={color == "primary" ? DarkLogo : LightLogo}
          alt="euka logo"
        />
      </a>
    </div>
  );
};

export default Logo;
