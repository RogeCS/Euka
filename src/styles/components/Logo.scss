@import "../global/variables.scss";

.logo {
  align-items: center;
  display: flex;
  justify-content: center;
}

.logo__image {
  height: 40px;
}
