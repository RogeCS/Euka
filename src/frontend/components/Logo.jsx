import React from "react";
import DarkLogo from "../assets/static/darkIcon.png";
import LightLogo from "../assets/static/lightIcon.png";

import "../styles/components/Logo.scss";

const Logo = ({ color }) => {
  return (
    <div className="logo">
      <a href="/" className="logo__link">
        <img
          className="logo__image"
          src={color == "primary" ? DarkLogo : LightLogo}
          alt="euka logo"
        />
      </a>
    </div>
  );
};

export default Logo;
