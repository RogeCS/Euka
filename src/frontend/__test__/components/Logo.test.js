import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Logo from "../../components/Logo.jsx";
import ProviderMock from "../../__mocks__/providerMock";

describe("<Logo />", () => {
  test("Logo image", () => {
    const logo = mount(
      <ProviderMock>
        <Logo />
      </ProviderMock>
    );
    expect(logo.find("img")).toHaveLength(1);
  });
  test("Logo snapshot", () => {
    const logo = create(
      <ProviderMock>
        <Logo />
      </ProviderMock>
    );
    expect(logo.toJSON()).toMatchSnapshot();
  });
});
