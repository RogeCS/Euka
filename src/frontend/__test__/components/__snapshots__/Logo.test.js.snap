// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> Logo snapshot 1`] = `
<div
  className="logo"
>
  <a
    className="logo__link"
    href="/"
  >
    <img
      alt="euka logo"
      className="logo__image"
      src="test-file-stub"
    />
  </a>
</div>
`;
